contracts/base/Escrow/Dapps/Uniswap.sol

```diff
@@ -18,75 +24,97 @@ import "./IDApp.sol";
 /**  Visit https://docs.openzeppelin.com/upgrades/2.6/proxies#upgrading-via-the-proxy-pattern for   **/
 /**  more information.                                                                              **/
 /*****************************************************************************************************/
-contract Uniswap is IDApp {
+/**
+    @notice This contract is used to define Uniswap dApp actions available. All dapp actions are invoked via 
+        delegatecalls from Escrow contract, so this contract's state is really Escrow.
+    @author develop@teller.finance
+ */
+contract Uniswap is IUniswap {
     using AddressLib for address;
+    using Address for address;
+
+    /* State Variables */
+    // State is shared with Escrow contract as it uses delegateCall() to interact with this contract.
+    
+    /**
+        @notice Swaps ETH/Tokens for Tokens/ETH using different Uniswap v2 Router 02 methods.
+        @param canonicalWeth address of the canonical WETH in the current network.
+        @param routerAddress address of the Uniswap Router v02.
+        @param path An array of token addresses. path.length must be >= 2. Pools for each consecutive pair of addresses must exist and have liquidity.
+        @param sourceAmount amount of source element (ETH or Tokens) to swap.
+        @param minDestination The minimum amount of output tokens that must be received for the transaction not to revert.
+        @dev This function mainly invokes 3 Uniswap external functions:
+            https://uniswap.org/docs/v2/smart-contracts/router02/#swapexactethfortokens
+            https://uniswap.org/docs/v2/smart-contracts/router02/#swapexacttokensforeth
+            https://uniswap.org/docs/v2/smart-contracts/router02/#swapexacttokensfortokens
+     */
+    function swap(
+        address canonicalWeth,
```

Comment:
Initially we had this as a constant but we need to change it on each network, so as it can't be a state variable in the dApp (delegatecall) we decided to pass it as a function parameter. Please feel free to suggest a different approach